### DISCUSSION TITLE
Integrating linting into Tailwind CLI (or a standalone tool)

### DISCUSSION THREAD

**Discussion by lostfictions** [tailwindlabs/tailwindcss#5698] [Ideas]
I love [Tailwind CSS IntelliSense](https://github.com/tailwindlabs/tailwindcss-intellisense). Being able to get autocomplete not just for Tailwind shortnames but also for the _longhand CSS properties they represent_ is a huge deal for me. Same with being able to see color swatches, size equivalents in multiple units -- the list goes on. I think it's probably no exaggeration to say the extension has significant bearing on how much you'll enjoy using Tailwind, especially if you're just starting out with it.

Another feature from the extension that I love is linting. It includes a bunch of useful warnings, including stuff like conflicting properties:

![Screenshot from 2021-10-05 02-37-13](https://user-images.githubusercontent.com/567041/135972803-e4129d9d-c4f6-46d0-9263-faf4008d0900.png)

One thing I've been wondering is: wouldn't it be great if these warnings were shown at build-time, too? It'd be great if an invocation of the `tailwind` CLI could show them alongside other info like the (soon to disappear) JIT preview warning. There could even be a mode like [Prettier's `--list-different`](https://prettier.io/docs/en/cli.html#--list-different), which would _only_ perform linting and exit with a nonzero status code if it detected any issues. That way we could easily integrate these lints into our CI systems.

I'm aware that [eslint-plugin-tailwindcss](https://github.com/francoismassart/eslint-plugin-tailwindcss) exists. It's wonderful that it does, but as far as I know it doesn't support the latest version of Tailwind, nor does it include all of the same lints. Running it alongside the Tailwind extension in VS Code often gives you duplicate warnings, and I've noticed it seems to slow down eslint for me. It's workable (for the lints it supports) but it seems like a lot of duplicated effort for a less-than-optimal development experience.

As Tailwind becomes more and more compiler-like -- especially with the (amazing) JIT engine set to fully replace the old one -- I think it'd be great if the default developer experience included some of the great lints that the Tailwind team has already created!

**Comment by elevatebart** [comment]
Very well put.
I know it is not an easy path, but there is way to do it.

For whomever needs a quick class sorter CLI in rust I found https://github.com/avencera/rustywind based on headwind.
